code/datums/components/jestosterone.dm

```diff
@@ -0,0 +1,43 @@
+/datum/component/jestosterone
+	var/mob_is_clown //Is the affected mob a clown?
+
+/datum/component/jestosterone/Initialize(mob/user, is_this_mob_a_clown)
+	mob_is_clown = is_this_mob_a_clown
+	if(mob_is_clown)
+		to_chat(user, "<span class='notice'>Whatever that was, it feels great!</span>")
+	else
+		to_chat(user, "<span class='warning'>Something doesn't feel right...</span>")
+	RegisterSignal(COMSIG_MOVABLE_MOVED, .proc/on_move)
+	RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/mob_was_attacked)
+	RegisterSignal(COMSIG_ITEM_ATTACK, .proc/mob_was_attacked)
+	RegisterSignal(COMSIG_ATOM_ATTACK_ALIEN, .proc/mob_was_attacked)
+	RegisterSignal(COMSIG_ON_MOB_LIFE, .proc/process_jestosterone)
```

Comment:
wait what

this should be in `on_mob_life` of the reagent in question.

Also, I'd advocate porting the `redirect` component, and putting these inside the `on_new` of the `jestosterone` reagent.

Then, in `on_mob_delete`, remove the redirect components from the mob.